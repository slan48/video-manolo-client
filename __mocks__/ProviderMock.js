import React from 'react';
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import {createWrapper, HYDRATE} from 'next-redux-wrapper';
import {reducer} from "../store";
import {Provider} from "react-redux";

const middleware = [thunk];

// create store
export const store = createStore(reducer, composeWithDevTools(applyMiddleware(...middleware)));

// create a makeStore function
const makeStore = context => store;

// export an assembled wrapper
export const wrapper = createWrapper(makeStore, {debug: process.env.NEXT_PUBLIC_ENV === 'development'});

const ProviderMock = (props) => (
  <Provider store={store}>
    {props.children}
  </Provider>
);

export default wrapper.withRedux(ProviderMock);
