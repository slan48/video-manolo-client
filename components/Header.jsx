import React from 'react';
import Link from "next/link";
import { useSelector } from "react-redux";
import Router from "next/router";

const Header = (props) => {
  const loggedIn = useSelector(state => state.loggedIn);

  return (
    <div className="container">
      <header className="animate__animated animate__fadeIn">
        <h1 onClick={() => Router.push('/')}>Video Manolo</h1>
        <ul>
          <li>
            <Link href="/">
              <a href="/">Home</a>
            </Link>
          </li>
          {!loggedIn ?
            <li>
              <Link href="/iniciar-sesion">
                <a href="/iniciar-sesion">Iniciar sesión</a>
              </Link>
            </li>
            :
            <li>
              <a href="/cerrar-sesion">Cerrar sesión</a>
            </li>
          }
        </ul>
        <style jsx>{`
        h1{
          cursor: pointer;
        }

        header{
          color: #fff;
          background: #000;  
          padding: 20px;
          display: flex;
          align-items: center;
          justify-content: space-between;
        }
        
        ul{
          list-style-type: none;
          display: flex;
          align-items: center;
          flex-wrap: wrap;
          padding: 0;
          margin: 0;
          
          a{
            padding: 0 20px;
            color: #fff;
          }
        }
      `}</style>
      </header>
    </div>
  );
};

export default Header;
