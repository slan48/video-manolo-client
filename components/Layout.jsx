import React from 'react';
import Header from "./Header";

const Layout = (props) => {
  return (
    <div className='main-layout'>
      <Header />
      {props.children}
    </div>
  );
};

export default Layout;
