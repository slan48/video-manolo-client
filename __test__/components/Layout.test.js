import React from 'react';
import { mount } from 'enzyme';
import Layout from "../../components/Layout";
import Header from "../../components/Header";
import ProviderMock from "../../__mocks__/ProviderMock";
import { create } from 'react-test-renderer';

describe('<Layout />', function () {
  const layout = mount(
    <ProviderMock>
      <Layout />
    </ProviderMock>
  );

  test('Layout component render', () => {
    expect(layout.length).toEqual(1);
  })

  test('Layout has header inside', () => {
    expect(layout.find('Header').length).toEqual(1);
  })

  test('Layout Snapshot', () => {
    const layout = create(
      <ProviderMock>
        <Layout />
      </ProviderMock>
    );
    expect(layout.toJSON()).toMatchSnapshot();
  })
});
