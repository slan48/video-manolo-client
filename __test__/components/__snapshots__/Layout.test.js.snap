// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Layout Snapshot 1`] = `
<div
  className="main-layout"
>
  <div
    className="container"
  >
    <header
      className="jsx-1681203433 animate__animated animate__fadeIn"
    >
      <h1
        className="jsx-1681203433"
        onClick={[Function]}
      >
        Video Manolo
      </h1>
      <ul
        className="jsx-1681203433"
      >
        <li
          className="jsx-1681203433"
        >
          <a
            className="jsx-1681203433"
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="jsx-1681203433"
        >
          <a
            className="jsx-1681203433 btn btn-light"
            href="/iniciar-sesion"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Iniciar sesi√≥n
          </a>
        </li>
      </ul>
    </header>
  </div>
</div>
`;
